### ISSUE TITLE
Issues with override when dir doesnt exist or when using USE statements

### ISSUE THREAD

**Issue by Prestaworks** [PrestaShop/PrestaShop#14948]
**Describe the bug**
If a module have overrides of controllers/front/listing these are not copies on install and not removed on uninstall.

**To Reproduce**
1. Install a module with a listing override.
2. Look in the overrides folder of Prestashop

**Additional information**
PrestaShop version: 1.7.6.0
PHP version: 7.2.x

**Comment by kpodemski** [comment]
What exactly do you want to override there? there's a very useful hook `filterContent` in every entity, maybe this would be enough?

**Comment by khouloudbelguith** [comment]
Hi @Prestaworks,

Could you please provide a demo module to check it.
In my case, I did not manage to reproduce the issue with PS1.7.6.0 using this module
[dummymodule.zip](https://github.com/PrestaShop/PrestaShop/files/3455850/dummymodule.zip)

Thanks!

**Comment by Prestaworks** [comment]
Hi Kholu,

See attached file.
[dummymodule.zip](https://github.com/PrestaShop/PrestaShop/files/3455867/dummymodule.zip)

**Comment by Prestaworks** [comment]
> 
> 
> What exactly do you want to override there? there's a very useful hook `filterContent` in every entity, maybe this would be enough?

I want to use my own search instead of the built in. So I need to override the listing controller so that Prestashop search provider is not used. I can't find any hook that can be used to replace the built in search and prevent not wanted DB queries.

**Comment by khouloudbelguith** [comment]
@Prestaworks, I tried to install the module provided => installed => OK, but it is not enabled.
I tried to enable it, but I have a red alert
```
Error!
Cannot enable module dummymodule. Unable to install override: directory (/projet/QA/1760fr/override/controllers/front/listing) not writable
```

https://drive.google.com/file/d/1VuYY9kzIB5HyopG_qg0dwhaqnRwBNzme/view

PS: permission is OK in my side: into the path /Project_Folder/override/controllers, I have all permissions:

```
drwxrwxrwx 3 khouloud.belguith khouloud.belguith 4096 juil. 24 08:25 admin
drwxrwxrwx 2 khouloud.belguith khouloud.belguith 4096 juil. 24 08:25 front
-rwxrwxrwx 1 khouloud.belguith khouloud.belguith 1316 juil. 10 17:07 index.php
```
I need to create the "Listing" folder manually 
Installing the module => ok => but it is disabled => override is not created
Enabling the module => OK => the override `SearchController.php` is created
Uninstalling the module => the override `SearchController.php` is not removed.

Same behavior with PS1.7.5.2.

Related to: #11516. 
I’ll add this to the debug roadmap so that it’s fixed. If you have already fixed it on your end or if you think you can do it, please do send us a pull request!
Thanks!

**Comment by Prestaworks** [comment]
I can add some more info after doing a lot of testing.
One of the issues is that the listing directory is not existing, so if it is created it will work (almost)

the other issue seems to be related to the "USE" statement, if this is removed from the override and the listing directory is created it will work.

So I think the module addOverride() needs to support to copy USE statements as well as being able to create the directory needed for the override.

**Comment by khouloudbelguith** [comment]
Hi,

I manage to reproduce the issue with the develop branch(1780).

Thanks!

**Comment by matks** [comment]
The problem with `use` has been referenced here https://github.com/PrestaShop/PrestaShop/issues/18592

**Comment by alisamie97** [comment]
Hi. I created a custom module which contains these files:
1. `override/controllers/front/AddressController.php`
2. `override/controllers/front/listing/CategoryController.php`

At first I only created the file `AddressController.php` and just added a new variable to check if it takes effect in root `override` folder when I reset the module. And this works properly.
Then I added the `CategoryController.php` and now my module has two override files. After resetting the module I checked the root `override` folder and there was no file for `CategoryController.php`, even the `AddressController.php` was missing. Also there was no errors and the reset showed me a success message.
I created directory of path `override/controllers/front/listing` in root of prestashop instance manually and reset the module again. This time I can see the overrides. I think this is not solved in Prestashop 1.7.8.6 yet. It just needs a listing folder.
